pandas/io/formats/html.py

```diff
@@ -264,12 +271,13 @@ def _write_header(self, indent):
                         values = (values[:ins_col] + [u('...')] +
                                   values[ins_col:])
 
-                name = self.columns.names[lnum]
-                row = [''] * (row_levels - 1) + ['' if name is None else
-                                                 pprint_thing(name)]
-
-                if row == [""] and self.fmt.index is False:
-                    row = []
+                row = [''] * (row_levels - 1)
```

Comment:
might be worth it to make this a function / method as it appears to repeat below.

can you add a comment on what this is doing

Reply:
the fixes for both the issues with the standard index and and issues with the multiindex have been purposefully made to use similar code to make the subsequent refactor simpler.

imo the refactor should create a `_get_formatted_column_labels` method for code parity with the `DataFrameFormatter` class but that refactor is outside the scope of this PR. the `DataFrameFormatter` class also has a `_get_column_name_list` which could be used.

However, i've not yet done this refactor because I want to get the bugs fixed first and the parametric tests in place for added assurance. I really wanted to limit the scope of this PR to the bug fixes since there is also a couple of issues with the header=False and sparsify=False which really need sorting before refactoring.

I will happily add a function but it would only be temporary. please let me know your thoughts.

Reply:
ok that’s fine for then
pls add comments though and a TODO here as well